tests/mitoai_ui_tests/agent.spec.ts

```diff
@@ -0,0 +1,138 @@
+import { expect, galata, test } from '@jupyterlab/galata';
+import {
+    createAndRunNotebookWithCells,
+    getCodeFromCell,
+    waitForIdle,
+} from '../jupyter_utils/jupyterlab_utils';
+import {
+    clickOnMitoAIChatTab,
+    clickPreviewButton,
+    clickAcceptButton,
+    sendMessageToMitoAI,
+} from './utils';
+
+test.describe("Agent mode integration tests", () => {
+
+    test.beforeEach(async ({ page }) => {
+        /*
+            Before each test, we switch to agent mode, and send a message. 
+        */
+
+        await createAndRunNotebookWithCells(page, []);
+        await waitForIdle(page);
+
+        await clickOnMitoAIChatTab(page);
+        await waitForIdle(page);
+
+        // Switch to agent mode
+        await page.getByRole('button', { name: 'Chat ▾' }).click();
+        await page.getByRole('button', { name: 'Agent' }).click();
+
+        await sendMessageToMitoAI(page, "print hi");
+        await waitForIdle(page);
+    });
+
+    test("Switch to agent mode, and send a message", async ({ page }) => {
+        // It's hard to know exactly how many items will be in the agent's plan,
+        // but we can assume there will be at least one message. 
+        const messageCount = await page.locator('.message-assistant-agent').count();
+        expect(messageCount).toBeGreaterThanOrEqual(1);
+    })
+
+    test("Edit message in agent's plan", async ({ page }) => {
```

Comment:
Can we also test that you can edit the original input before pressing Lets Go and it will regenerate the plan